pandas/core/arrays/numpy_.py

```diff
@@ -355,9 +355,14 @@ def to_numpy(  # type: ignore[override]
         copy: bool = False,
         na_value=lib.no_default,
     ) -> np.ndarray:
-        result = np.asarray(self._ndarray, dtype=dtype)
+        temp = self
 
-        if (copy or na_value is not lib.no_default) and result is self._ndarray:
+        if na_value is not lib.no_default:
+            temp.fillna(method="bfill")
```

Comment:
Why bfill? Also, this line won't have any effect anyway because `inplace=False` by default.

Reply:
And `na_value` is no longer being used to fill